src/inc/clrconfigvalues.h

```diff
@@ -1010,6 +1010,10 @@ RETAIL_CONFIG_DWORD_INFO(EXTERNAL_ReadyToRun, W("ReadyToRun"), 1, "Enable/disabl
 RETAIL_CONFIG_DWORD_INFO(EXTERNAL_ReadyToRun, W("ReadyToRun"), 0, "Enable/disable use of ReadyToRun native code") // Off by default for desktop
 #endif
 
+#if defined(FEATURE_PAL) && defined(FEATURE_EVENT_TRACE)
```

Comment:
With the env. var, I think you need to validate that there is always a mechanism to pass it to the host that actually starts the runtime. If the host is contained within a service/daemon, I am not sure if the env var can be passed to it.

IMHO, the ability to enable/disable tracing can be set via a GUI/Configuration page and persisted to a file that a native host can read and pass it to CLR startup flags. This approach would work for services/daemons too.

Your approach works well for simple, explicity invoked hosts but not the more involved ones.